### PR TITLE
build: use built-in formatter

### LINKED ISSUE #1274: Replace outdated tool dotnet-format by the built-in dotnet format
> BTW, it seems we are using an outdated tool dotnet-format instead of the built-in dotnet format subcommand in CI. The former seems to be completely confused by raw string literals.

_Originally posted by @Oberon00 in https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/1270#issuecomment-1640462239_

Similar changes done in autoinstrumentation repository https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2694 and 
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2614/files

### PR DESCRIPTION
**Pull request by BurakTaner**
Fixes #1274.

## Changes
I changed outadated formatter with built-in one.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by Kielek** [comment]
@BurakTaner,
please sign EasyCLA before we can proceed.
I suppose that it will detect some issues which we need to fix together with this merge.

**Comment by BurakTaner** [comment]
Hi @Kielek, I tried to sign EasyCLa with your email and name-surname but gave me error "Company not found". My first time with EasyCLA, can you give a little help?

**Comment by Kielek** [comment]
@BurakTaner, you have two options
1. Register as individual contributor
2. Register as a part of the company were you working on. It depends on when you want to contribute  to OTel repositories. In this case you have to find manager in your company to proceed.

**Comment by BurakTaner** [comment]
@Kielek I signed CLA and pushed last changes to make branch up to date with main. What should ı do next?

File: .github/workflows/dotnet-format.yml [hunk 1]
```diff
@@ -29,4 +29,4 @@ jobs:
       run: dotnet tool install -g dotnet-format
```
Comment (utpilla) [inline .github/workflows/dotnet-format.yml:29]: If it's part of the newer .NET SDK, then do we still need to install `dotnet-format`?

File: .github/workflows/dotnet-format.yml [hunk 2]
```diff
@@ -29,4 +29,4 @@ jobs:
...
 
     - name: dotnet format
-      run: dotnet-format --folder --check
+      run: dotnet format .\opentelemetry-dotnet-contrib.sln --no-restore --verify-no-changes
```
Comment (utpilla) [inline .github/workflows/dotnet-format.yml:32]: Do we need to specify the `.sln` file?

**Comment by utpilla** [comment]
Since this PR does not change any `.cs` file or `.editorconfig`, the dotnet format check is not being triggered. We need to ensure that this check works correctly before we merge this PR. You might have to update a `.cs` file and then check the dotnet-format run, before undoing the change to the `.cs` file.

**Comment by Kielek** [comment]
@BurakTaner, I have pushed two commit to your branch
1. typo fix to execute real dotnet-format build
2. drop installation for obsolete tool.

Please check what you can do with detected issues https://github.com/open-telemetry/opentelemetry-dotnet-contrib/actions/runs/5676698870/job/15383851038?pr=1276

**Comment by BurakTaner** [comment]
Branch was not up to date with the current changes and ı accidently. That's remove your commits, ı am so sorry about it. Going to make those changes again.

File: .github/workflows/dotnet-format.yml (hunk 2 above)
Comment (utpilla) [reply .github/workflows/dotnet-format.yml:29]: Could you simplify this command by removing the parameter for providing the `.sln` file?

**Comment by BurakTaner** [comment]
@Kielek @utpilla I tried to use format without errors in my machine and it's only works if ı build before the format process. The old one (dotnet-format) don't need this. Idk what is related to and how could ı fix it. Should ı add a build step in yml or are we going to use the old one ?

**Comment by Kielek** [comment]
I see a lot of issues similar to 

```
Required references did not load for OpenTelemetry.Contrib.Tests.Shared or referenced project. Run `dotnet restore` prior to formatting.
```

I think that you have 2 options to proceed
1. Add separate step which will restore packages before current one
2. Remove `--no-restore` from current command.

I think the first option should be better to have separate time measurement for just code formatting.

Keeping the old version is not good idea - it is not supporting new C# futures so it will prevent us using it.

**Comment by BurakTaner** [comment]
@Kielek Like ı said, ı tried to use format in my computer and without dotnet build, still gave the error like the last checks. Should ı include a dotnet build step or is there any problem am ı missing because ı already added restore step to yml file?

**Pull request closed without merging**